tests/core/contracts/test_contract_call_interface.py

```diff
@@ -612,3 +617,61 @@ def test_invalid_fixed_value_reflections(web3, fixed_reflection_contract, functi
     contract_func = fixed_reflection_contract.functions[function]
     with pytest.raises(ValidationError, match=error):
         contract_func(value).call({'gas': 420000})
+
+
+@pytest.mark.parametrize(
+    'method_input, expected',
+    (
+        (
+            {'a': 123, 'b': [1, 2], 'c': [
+                {'x': 234, 'y': [True, False], 'z': [
+                    '0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                ]},
+                {'x': 345, 'y': [False, False], 'z': [
+                    '0xefd1FF70c185A1C0b125939815225199079096Ee',
+                    '0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e',
+                ]},
+            ]},
+            (123, [1, 2], [
+                (234, [True, False], [
+                    '0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                ]),
+                (345, [False, False], [
+                    '0xefd1FF70c185A1C0b125939815225199079096Ee',
+                    '0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e',
+                ]),
+            ]),
+        ),
+        (
+            (123, [1, 2], [
+                (234, [True, False], [
+                    '0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                ]),
+                (345, [False, False], [
+                    '0xefd1FF70c185A1C0b125939815225199079096Ee',
+                    '0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e',
+                ]),
+            ]),
+            (123, [1, 2], [
+                (234, [True, False], [
+                    '0x4AD7E79d88650B01EEA2B1f069f01EE9db343d5c',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                    '0xfdF1946A9b40245224488F1a36f4A9ed4844a523',
+                ]),
+                (345, [False, False], [
+                    '0xefd1FF70c185A1C0b125939815225199079096Ee',
+                    '0xf35C0784794F3Cd935F5754d3a0EbcE95bEf851e',
+                ]),
+            ]),
+        ),
+    ),
+)
```

Comment:
It looks like this is already working, but might be nice to add a regression test for nested arrays of tuples. Seems like the kind of thing that could break in a rewrite/refactor.